### DISCUSSION TITLE
can add additional column for example user_id

### DISCUSSION THREAD

**Discussion by hossein-holy** [spatie/laravel-tags#475] [Ideas]
Hi, your package is good package, but can add  additional column for example `user_id` .
assume we have `Product` model and every user can select own tags then in your function should send additional parameter like:
`$product->attachTag($tagName, $type, array additionalData);`
`$product->attachTag('tag 1', 'example', ['user_id' => auth()->id()]);`

**Comment by MattiaMarchiorato** [comment]
@hossein-holy we have the same problem, for now we just add a migrtion to add to the tags table.

**Reply by HusseinIbrahem** [reply]
and how to fill the new column?

**Reply by MattiaMarchiorato** [reply]
You can use your tag model 
https://spatie.be/docs/laravel-tags/v3/advanced-usage/using-your-own-tag-model
and set the logic you need in this model, at the end is a polymorphic relation, at the end we implemented our tag functionality ;)
